src/core/include/iDynTree/Core/TestUtils.h

```diff
@@ -311,6 +311,46 @@ namespace iDynTree
         }
     }
 
+    /**
+     * Assert that two vectors are equal, and exit with EXIT_FAILURE if they are not.
+     *
+     * The tolerance passed in this function is a relative tolerance on the max element of the comparison, i.e.
+     * absoluteTol = max(relativeTol*max(val1,val2), minAbsoluteTol)
+     */
+    template<typename VectorType1, typename VectorType2>
+    void assertVectorAreEqualWithRelativeTol(const VectorType1& vec1, const VectorType2& vec2, double relativeTol, double minAbsoluteTol, std::string file, int line)
+    {
+        if( vec1.size() != vec2.size() )
+        {
+            std::cerr << file << ":" << line << " : assertVectorAreEqualWithRelativeTol failure: vec1 has size " << vec1.size()
+                      << " while vec2 has size " << vec2.size() << std::endl;
+            exit(EXIT_FAILURE);
+        }
+
+        std::vector<bool> correctElements(vec1.size(),true);
+        bool checkCorrect = true;
+
+        for( unsigned int i = 0; i < vec1.size(); i++ )
+        {
+            double absoluteTol = std::max(relativeTol*std::max(std::abs(vec1(i)), std::abs(vec2(i))), minAbsoluteTol);
+            if( fabs(vec1(i)-vec2(i)) >= absoluteTol )
+            {
+                checkCorrect = false;
+                correctElements[i] = false;
+            }
+        }
+
+        if( !checkCorrect )
+        {
+            std::cerr << file << ":" << line << " : assertVectorAreEqualWithRelativeTol failure: " << std::endl;
+            printVector("vec1",vec1);
```

Comment:
Comment for a later PR: I think we could add a newline between the various prints, so we understand when a vector ends and the other starts

Reply:
Yes, I am also always bothered by that.